### ISSUE TITLE
Exception on Neo4j.Driver.Internal.Connector.ChunkedInputStream.ReadSpecifiedSize- "Expected X got Y"

### ISSUE THREAD

**Issue by sachafaust** [neo4j/neo4j-dotnet-driver#88]
Fairly persistent issue but sometime it works without problems

Exception - "Error - Expect 471, but got 265"  where both # changes depending when I run the query. Using Driver v1 with Neo4j 3.0.1

**[My code]**
            using (var driver =
                GraphDatabase.Driver(
                    "bolt://server",
                    GetAuthToken()))
            {
                using (var session = driver.Session())
                {
                    try
                    {

```
                    var result = session.Run(query);  <-- Fails here
```

**Query**
match (admins:User)-[:SERVICE_ADMIN]->(service:Service)-[:RESOURCES]->(f:File)
WITH service, collect(f.name) as files, collect(admins.accountname) as admins
return service.name, admins, files;

**Stacktrace**
   at **Neo4j.Driver.Internal.Connector.ChunkedInputStream.ReadSpecifiedSize(Byte[] buffer)** in C:\BuildAgent\work\9ae188c903d07190\Neo4j.Driver\Neo4j.Driver\Internal\Connector\ChunkedInputStream.cs:line 131
   at Neo4j.Driver.Internal.Connector.ChunkedInputStream.Ensure(Int32 size) in C:\BuildAgent\work\9ae188c903d07190\Neo4j.Driver\Neo4j.Driver\Internal\Connector\ChunkedInputStream.cs:line 114
   at Neo4j.Driver.Internal.Connector.ChunkedInputStream.ReadByte() in C:\BuildAgent\work\9ae188c903d07190\Neo4j.Driver\Neo4j.Driver\Internal\Connector\ChunkedInputStream.cs:line 52
   at Neo4j.Driver.Internal.Packstream.PackStream.Unpacker.UnpackStructHeader() in C:\BuildAgent\work\9ae188c903d07190\Neo4j.Driver\Neo4j.Driver\Internal\Packstream\PackStream.cs:line 556
   at Neo4j.Driver.Internal.Packstream.PackStreamMessageFormatV1.ReaderV1.Read(IMessageResponseHandler responseHandler) in C:\BuildAgent\work\9ae188c903d07190\Neo4j.Driver\Neo4j.Driver\Internal\Packstream\PackStreamMessageFormatV1.cs:line 54
   at Neo4j.Driver.Internal.Connector.SocketClient.Receive(IMessageResponseHandler responseHandler) in C:\BuildAgent\work\9ae188c903d07190\Neo4j.Driver\Neo4j.Driver\Internal\Connector\SocketClient.cs:line 107
   at Neo4j.Driver.Internal.Connector.SocketConnection.Sync() in C:\BuildAgent\work\9ae188c903d07190\Neo4j.Driver\Neo4j.Driver\Internal\Connector\SocketConnection.cs:line 68
   at Neo4j.Driver.Internal.Session.<>c__DisplayClass8_0.<Run>b__0() in C:\BuildAgent\work\9ae188c903d07190\Neo4j.Driver\Neo4j.Driver\Internal\Session.cs:line 83
   at Neo4j.Driver.Internal.LoggerBase.TryExecute[T](Func`1 func) in C:\BuildAgent\work\9ae188c903d07190\Neo4j.Driver\Neo4j.Driver\Internal\LoggerBase.cs:line 54

**Comment by zhenlineo** [comment]
I am so sorry about this error. It has just been fixed in https://github.com/neo4j/neo4j-dotnet-driver/pull/84
I will try to release a new version this week to include the fix for it. 
You might try to use the snapshot feed for now
`PM> Install-Package Neo4j.Driver -Version 1.0.0-SNAPSHOT-221`
or 
`PM> Install-Package Neo4j.Driver -Version 1.1.0-SNAPSHOT-87`
(1.1 and 1.0 are the same till now)

Also let me know if you still have the same problem with the snapshot (the faster to confirm the fix, the faster to get a new release :P)

**Comment by sachafaust** [comment]
Thank you for the prompt reply/fix. Unfortunately I get the following when trying to get the snapshot. Using Visual studio 2015. 

Install-Package : Package 'Neo4j.Driver' is not found
At line:1 char:1
- Install-Package Neo4j.Driver -Version 1.1.0-SNAPSHOT-87
- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  - CategoryInfo          : NotSpecified: (:) [Install-Package], Exception

**Comment by zhenlineo** [comment]
Have you add our myget feed into your VS? The snapshot feed was in the readme. Check this page https://myget.org/gallery/neo4j-driver-snapshots

**Comment by zhenlineo** [comment]
Pls use snapshot-87 or snapshot-yymmddhhmm. As you could see, the numbering was wrong and I changed to date numbering yesterday. The old numbering artifacts will go away after a few days.

**Comment by sachafaust** [comment]
Fix confirmed thank you. Any ETA on when this will be part of the main release?

**Comment by zhenlineo** [comment]
Released [1.0.2](https://www.nuget.org/packages/Neo4j.Driver/)

**Comment by zhenlineo** [comment]
I am closing this issue as the fix is already merged and released.
